include/odp/api/spec/traffic_mngr.h

```diff
@@ -331,6 +331,52 @@ typedef struct {
 	/** The per_level array specifies the TM system capabilities that
 	 * can vary based upon the tm_node level. */
 	odp_tm_level_capabilities_t per_level[ODP_TM_MAX_LEVELS];
+
+	/** dynamic_topology_update indicates support for TM system dynamic
+	 * topology update. A dynamic topology update is defined as update to
+	 * a TM system topology while TM system is not in stopped state.
+	 * When TRUE, application can update topology dynamically
+	 * without bringing the TM system to stopped state. When FALSE,
+	 * application has to call odp_tm_stop() before updating the
+	 * topology and odp_tm_start() after completing the update.
+	 */
+	odp_bool_t dynamic_topology_update;
+
+	/** dynamic_shaper_update indicates support for TM system's dynamic
+	 * shaper profile changes. When TRUE, application can update shaper
+	 * profile of a TM queue or TM node dynamically.
+	 * When FALSE, it implies that TM system should be brought to
+	 * stopped state before doing any TM node or TM queue shaper profile
+	 * change.
```

Comment:
I wonder if it is clear that the lack of dynamic update capability means that the shaper of a node 0or queue cannot be touched in either of the two ways that are possible (odp_tm_{queue,node}_shaper_config() and odp_tm_shaper_params_update()). Maybe the end of the last sentence could be rephrased like this "...before changing the shaper profile or updating the parameters of the shaper profile of any TM node or TM queue."

And the same comment for the other similar capability fields.